extensions/functions_geometry.yaml

```diff
@@ -0,0 +1,39 @@
+%YAML 1.2
+---
+scalar_functions:
+  -
+    name: "st_point"
+    description: >
+      Return a point with the given x and y coordinate values.
+    impls:
+      - args:
+          - name: x
+            value: fp32
+          - name: y
+            value: fp32
+        return: point
+  -
+    name: "st_makeline"
+    description: >
+      Return a linestring containing the points of point, multipoint, or linestring geometries.
+      Repeated points at the beginning of input linestring are collapsed to a single point.
+      Repeated points in point and multipoint inputs are not collapsed.
+    impls:
+      - args:
+          - name: x
+            value: point
+          - name: y
+            value: point
+        return: linestring
+      - args:
+          - name: x
+            value: multipoint
+          - name: y
+            value: multipoint
+        return: linestring
+      - args:
+          - name: x
+            value: linestring
```

Comment:
What are the semantics here?  Do we append the two lines assuming there is a join point or does it create a line between x's endpoint and y's startpoint?

Reply:
It doesn't necessarily have to join.  We can form an open (or simple) linestring, with 2 linestrings that don't connect.

http://postgis.net/workshops/postgis-intro/geometries.html#linestrings

Reply:
The semantics would be useful to have in the description.

Reply:
That's helpful.  The part I was seeking clarification on was whether call st_makeline with two linestrings has any special behavior.  Are the endpoints connected or not as a result of this operation?   It appears that the two liststrings (effectively list<list<point>>) are just concatenated.  To connect the two list strings one would need to add the first new list<point> to the last list<point> and append the rest.